lib/types/operation.js

```diff
@@ -423,55 +430,91 @@ Operation.prototype.getHeaderParams = function (args) {
   return headers;
 };
 
-Operation.prototype.urlify = function (args) {
-  var formParams = {};
-  var requestUrl = this.path;
-  var querystring = ''; // grab params from the args, build the querystring along the way
+Operation.prototype.rfc6570CompliantUrl = function(path) {
+  var queryStringVariables = '';
+  var noQueryParamsPath = this.stripQueryStringVariables(path);
+  var queryParams = _.filter(this.parameters, function (param) {
+    return param.in === 'query';
+  });
+  if (queryParams.length > 0) {
+      if (noQueryParamsPath.indexOf('?') >= 0) {
+        queryStringVariables = '{&';
+      } else {
+        queryStringVariables = '{?';
+      }
+  }
 
-  for (var i = 0; i < this.parameters.length; i++) {
-    var param = this.parameters[i];
+  var queryParamCount = 0;
+  for (var i = 0; i < queryParams.length; i++) {
+    var param = queryParams[i];
 
-    if (typeof args[param.name] !== 'undefined') {
-      if (param.in === 'path') {
-        var reg = new RegExp('\{' + param.name + '\}', 'gi');
-        var value = args[param.name];
+    if (param.in === 'query') {
+      if (queryParamCount > 0) {
+          queryStringVariables += ',';
+      }
+      var paramSuffix = '';
+      if (param.collectionFormat === 'multi') {
+        paramSuffix = '*';
+      }
+      queryStringVariables += this.encodeQueryParam(param.name) + paramSuffix;
+      queryParamCount++;
+    }
+  }
 
-        if (Array.isArray(value)) {
```

Comment:
I don't think it's safe to get rid of this code.  When interacting directly with the javascript client, you can pass arrays directly instead of encoding them into strings